lib/gooddata/models/project.rb

```diff
@@ -1757,7 +1758,8 @@ def remove_users(list)
     end
 
     def check_groups(specified_groups, options = {})
-      groups = user_groups.map(&:name)
+      @user_groups_cache = user_groups if @user_groups_cache.nil?
```

Comment:
@user_groups_cache is never invalidated which makes the `check_groups` method unreliable. Can we work around this by passing the "cache" in options instead of keeping it as an instance variable?